bot/cogs/moderation/modlog.py

```diff
@@ -116,6 +118,21 @@ def ignore(self, event: Event, *items: int) -> None:
 
         return await self.bot.get_context(log_message)  # Optionally return for use with antispam
 
+    async def reupload_attachments(
+            self,
+            message: discord.Message,
+            channel_id: int = GuildConstant.attachment_repost
+    ) -> t.List[str]:
+        """Re-upload message's attachments to the the channel_id and return the list of re-posted attachments URLs."""
+        channel = self.bot.get_channel(channel_id)
+        out = []
+        for attachment in message.attachments:
+            buffer = BytesIO()
+            await attachment.save(buffer, use_cached=True)
+            reupload = await channel.send(file=discord.File(buffer, filename=attachment.filename))
+            out.append(reupload.attachments[0].url)
```

Comment:
You can use `ContextManager` here to ensure `buffer` will close properly, like this
```py
            with BytesIO() as buffer:
                await attachment.save(buffer, use_cached=True)
                reupload: discord.Message = await channel.send(file=discord.File(buffer, filename=attachment.filename))
                out.append(reupload.attachments[0].url)
```